docs/src/hooks/useLockedBody.ts

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from 'react';
+import useIsomorphicLayoutEffect from './useIsomorphicLayoutEffect';
+
+type UseLockedBodyOutput = [boolean, (locked: boolean) => void];
+
+function useLockedBody(
+  initialLocked = false,
+  rootId = 'root'
+): UseLockedBodyOutput {
+  const [locked, setLocked] = useState(initialLocked);
+
+  // Do the side effect before render
+  useIsomorphicLayoutEffect(() => {
+    if (!locked) {
+      return;
+    }
+
+    // Save initial body style
+    const originalOverflow = document.body.style.overflow;
+    const originalPaddingRight = document.body.style.paddingRight;
+
+    // Lock body scroll
+    document.body.style.overflow = 'hidden';
+
+    // Get the scrollBar width
+    const root = document.getElementById(rootId);
+    const scrollBarWidth = root ? root.offsetWidth - root.scrollWidth : 0;
+
+    // Avoid width reflow
+    if (scrollBarWidth) {
+      document.body.style.paddingRight = `${scrollBarWidth}px`;
+    }
+
+    return () => {
+      document.body.style.overflow = originalOverflow;
+
+      if (scrollBarWidth) {
+        document.body.style.paddingRight = originalPaddingRight;
+      }
+    };
+  }, [locked]);
+
+  // Update state if initialValue changes
+  useEffect(() => {
+    if (locked !== initialLocked) {
+      setLocked(initialLocked);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Spent some time looking at this hook and I'm not a fan of this `useEffect`. Not listing the deps, especially `locked` and `setLocked` could cause some possible bugs. 

I'd suggest removing this whole useEffect as it is listening for changes to `initialValue` and updates the state, but if it is doing that it is no longer 'initialValue' any more, it is just regular state! 

We aren't changing this value dynamically anyway, so safe to remove without causing any issues for our current use-case.

Reply:
Ahh great catch! And thanks for the explanation <3